apps/bare-expo/scripts/start-ios-e2e-test.ts

```diff
@@ -157,6 +157,45 @@ export function setupLogger(predicate: string, signal: AbortSignal): () => Promi
   };
 }
 
+async function startSimulatorAsync(deviceId: string, timeout: number = 1000 * 60 * 3) {
+  await retryAsync(async (retryNumber) => {
+    if (process.env.CI) {
+      try {
+        await spawnAsync('xcrun', ['simctl', 'shutdown', deviceId], { stdio: 'inherit' });
+        await spawnAsync('xcrun', ['simctl', 'erase', deviceId], { stdio: 'inherit' });
+      } catch {}
+    }
+
+    console.time(
+      `\n📱 Starting Device - name[${TARGET_DEVICE}] udid[${deviceId}] retry[${retryNumber}]`
+    );
+    const bootProc = spawnAsync('xcrun', ['simctl', 'bootstatus', deviceId, '-b'], {
+      stdio: 'inherit',
+    });
+
+    let timeoutHandle: NodeJS.Timeout | null = null;
+    const timeoutPromise = new Promise((_, reject) => {
+      timeoutHandle = setTimeout(() => {
+        bootProc.child?.kill('SIGTERM');
+        reject(new Error('Timeout from booting up simulator'));
+      }, timeout);
+    });
+
+    await Promise.race([bootProc, timeoutPromise]);
+    if (timeoutHandle) {
+      clearTimeout(timeoutHandle);
+      timeoutHandle = null;
+    }
+
+    await spawnAsync('open', ['-a', 'Simulator', '--args', '-CurrentDeviceUDID', deviceId], {
```

Comment:
I wonder, (why) do we actually need to start the simulator?

Reply:
it's mostly for real users running this script. for `xcrun simctl boot` (or `xcrun simctl bootstatus -b`), it only boots the simulators in background. people cannot see the simulators in foreground. the open Simulator.app takes it from background to foreground. we also do this in expo cli fwiw.